### PR TITLE
fix: "Free text import does not recognize new TLDs", issue #1149

### PR DESCRIPTION
**Pull request by cristianbell**
#### What does it do?

`fixes #<1149>`
#### Questions
- [ ] Does it require a DB change?
- [ ] Are you using it in production?
- [ ] Does it require a change in the API (PyMISP for example)?
#### Release Type:
- [ ] Major
- [X] Minor
- [ ] Patch

### CODE REVIEW

**Comment by rotanid** [comment]
could you please check your indention, if you look at the diff here on github you notice that many changes aren't changes in the code but only in the indention. also part of the new code has wron indention.
(thanks, you also fixed some indention and obsolet spaces in this PR, which is OK in this case because these were only in the same parts of the code that got changed/added.)

File: app/Controller/EventsController.php [hunk 1]
```diff
@@ -2802,9 +2802,15 @@ public function freeTextImport($id) {
 		}
 
 		if ($this->request->is('post')) {
+            //getting the warninglists
+            App::uses('WarninglistsController', 'Controller');
+            $warningList = new WarninglistsController();
```
Comment (iglocska) [inline app/Controller/EventsController.php]: You should never have to call functions from a different controller, this breaks basic MVC principles.

**Comment by cristianbell** [comment]
should be ok now, please take a look when u have time. thanks!

File: app/Controller/EventsController.php [hunk 2]
```diff
@@ -2804,7 +2804,10 @@ public function freeTextImport($id) {
 		if ($this->request->is('post')) {
 			App::uses('ComplexTypeTool', 'Tools');
 			$complexTypeTool = new ComplexTypeTool();
-			$resultArray = $complexTypeTool->checkComplexRouter($this->request->data['Attribute']['value'], 'FreeText');
+			$this->loadModel('Warninglist');
+			$IANATLDentries = $this->Warninglist->getAllIANAEntries();
+
+            $resultArray = $complexTypeTool->checkComplexRouter($this->request->data['Attribute']['value'], 'FreeText', $IANATLDentries);
```
Comment (Rafiot) [inline app/Controller/EventsController.php]: This line is wrongly indented, I think it is the last one.

**Comment by cristianbell** [comment]
and that one fixed as well, thanks.

**Comment by iglocska** [comment]
I just reviewed the latest version and here are the issues I found:

1 (major).
Currently if the warninglists are not updated on a blank installation this fails with:

Notice (8): Undefined index: WarninglistEntry [APP/Model/Warninglist.php, line 243]
Warning (2): array_map() [function.array-map]: Argument #2 should be an array [APP/Model/Warninglist.php, line 243]

It should check whether the warninglist exists instead of assuming that it does. Not finding the correct warninglist should make the freetext import fall back to the old functinality.

2 (critical). 
Once I enabled the warninglist, resolving a domain name correctly identifies it as network activity, yet the type field is broken/missing. I didn't go through the code what change causes this but this breaks the core functionality of the freetext import tool as well as most likely the enrichment and import modules as those rely on the same functionality internally. 

3 (cosmetic, worst case leave this I'll fix it if the rest works). 
Coding style issues: Spacing around conditionals, loops as well as non-brace-encapsulated contents thereof.

**Comment by cristianbell** [comment]
will create a new pull request and work on it w/ @cvandeplas

**Pull request closed without merging**